File: 8KYU/Collatz Conjecture (3n+1)/solution.clj
(ns clojure.collatz)
(defn hotpo [n]
  ((fn [n c] 
    (cond(= 1 n)         c
         (= 0 (mod n 2)) (recur (/ n 2) (inc c))      
         :else           (recur (inc (* n 3)) (inc c)))) n 0))

File: 8KYU/Collatz Conjecture (3n+1)/solution.js
var hotpo = function(n){
    var c = 0;
  
    while(n != 1) {
      c++;
      if (n % 2 == 0) {
         n /= 2;
      }
      else {
        n = 3 * n + 1;
      }
    }
  
    return c;
    
}